packages/livekit-rtc/src/e2ee.ts

```diff
@@ -259,7 +259,8 @@ export class E2EEManager {
     const options = { ...defaultE2EEOptions, ...opts };
 
     this.options = options;
-    this.keyProvider = new KeyProvider(roomHandle, options.keyProviderOptions);
+    this.keyProvider =
+      options.keyProviderOptions && new KeyProvider(roomHandle, options.keyProviderOptions);
```

Comment:
but this line here sets keyProvider to false/undefined if there are no keyProviderOptions passed. That doesn't seem right. There should always be a key provider, even if it's initialized with the default options

Reply:
ah, i see. the default values were ignored outside of this function. latest commit runs the e2ee options through `...defaultE2EEOptions,` before continuing in RoomOptions